py3status/modules/diskdata.py

```diff
@@ -87,17 +105,73 @@ def post_config_hook(self):
             value - value (float)
             unit - unit (string)
         """
-        self.last_diskstats = self._get_diskstats(self.disk)
+        types_info = {
+            "disks": {"major": [3, 8, 22]},
+            "raid": {"major": [9]},
+            "cdrom": {
+                "major": [
+                    11,
+                    15,
+                    16,
+                    17,
+                    18,
+                    20,
+                    23,
+                    24,
+                    25,
+                    26,
+                    27,
+                    28,
+                    29,
+                    30,
+                    32,
+                    33,
+                    34,
+                ]
+            },
+            "mmc": {"major": [31]},
+        }
+
+        self._disks = {}
+
+        if not self.disk:
+            allowed_types = []
+            for _type in self.types.split():
+                allowed_types.extend(types_info[_type]["major"])
+            with open("/proc/diskstats") as ds:
+                for stat in ds:
+                    line = stat.split()
+                    if int(line[0]) in allowed_types and int(line[1]) == 0:
```

Comment:
Is this correct? `int(line[1]) == 0:` == `Minor version`... to block `dm-2`, `loop*`, `sr0`, etc?

Reply:
Kept this design decision from current version. The whole "types" thing is a bit fuzzy ATM, and I'm wondering if it worth the effort at all...

Reply:
I've removed the filter on non minor-0 devices